pages/pipelines/conditionals.md.erb

```diff
@@ -0,0 +1,119 @@
+# Conditionals
+
+- the `if` parameter provides the ability to conditionally run steps
+- using c-like expressions, boolean conditions are defined to restrict the running of steps to specific circumstances
+- can be used in all YAML-based pipelines
+
+<%= toc %>
+
+## `if` Syntax
+
+- use the `if` param to conditionally run a step
+- the below example will only run the `tests` step if the build message doesn't contain the string "skip me"
+
+```
+steps:
+	- command: ./scripts/tests.sh
+	  label: tests
+	  - if:
+	  	build.message != "skip me"
+```
+
+- use it in any step type
+- any of the supported expressions and parameters
+- cannot be used at the same time as the `branch` parameter
+
+## Supported Expressions
+
+ - Comparators: == != =~ !~
```

Comment:
In some languages, regular expressions can be on either side of an operator, but in our language, they can only be on the right hand side. i.e. this is totally valid ruby:

```ruby
irb(main):006:0> "hello" =~ /hello/
=> 0
irb(main):007:0> /hello/ =~ "hello"
=> 0
```

But for us, you can only do write it like this:

```
"hello" =~ /hello/
```